### PR TITLE
Feat/message bar

### PR DESCRIPTION
**Pull request by kevodwyer**
This turns the existing feedback functionality into a means of providing user <--> server messaging.

### CODE REVIEW

**Comment by kevodwyer** [comment]
messages

**Comment by kevodwyer** [comment]
feedback

**Comment by kevodwyer** [comment]
messages

File: src/components/feedback/feedback.html [hunk 1]
```diff
@@ -3,19 +3,67 @@
     <meta http-equiv="content-type" content="text/html; charset=utf-8" />
     <div style="height:30%"></div>
     <div class="modal-container" @click.stop style="height:70%;overflow-y:auto">
+        <span @click="close" class="close">&times;</span>
         <div class="modal-header">
-            <h2>Feedback</h2>
+            <div v-if="isFeedback" >
```
Comment (ianopolous) [inline src/components/feedback/feedback.html:8]: using feedback component for messages as well feels a little wrong to me. If we're going to do it this way, then we should generify it a bit more. So have the title as a property for example. Essentially try and remove all the isFeedback branches.

File: src/components/filesystem/index.js [hunk 2]
```diff
@@ -196,14 +199,43 @@ module.exports = {
             }
         });
             }
+        this.showPendingMessages();
 	},
-	
-        processPending: function() {
-            for (var i=0; i < this.onUpdateCompletion.length; i++) {
-                this.onUpdateCompletion[i].call();
+	showPendingMessages: function() {
```
Comment (ianopolous) [inline src/components/filesystem/index.js:204]: can we call this showPendingServerMessages? I'm trying to keep the naming clear and reserve message on its own for keymail.

File: src/components/filesystem/index.js [hunk 3]
```diff
@@ -887,6 +919,144 @@ module.exports = {
...
+                that.showSpinner = false;
+            });
+        },
+
+        sendMessage: function(msgId, contents) {
+            let that = this;
+            let message = this.getMessage(msgId);
+            if (message != null) {
+                this.showSpinner = true;
+                var trimmedContents = contents.length > 1000 ? contents.substring(0, 1000) : contents;
```
Comment (ianopolous) [inline src/components/filesystem/index.js:973]: we define max length in the ServerMessage class. Can we reference that directly? Does it need a jsannotation?

Comment (kevodwyer) [reply src/components/filesystem/index.js]: done

File: src/components/filesystem/index.js (hunk 2 above)
Comment (kevodwyer) [reply src/components/filesystem/index.js]: done

File: src/components/feedback/feedback.html (hunk 1 above)
Comment (kevodwyer) [reply src/components/feedback/feedback.html]: done

**Pull request merged**